### ISSUE TITLE
[Project Manager] Support for multi-project apps

### ISSUE THREAD

**Issue by wojciech-kulik** [wojciech-kulik/xcodebuild.nvim#153]
### Discussed in https://github.com/wojciech-kulik/xcodebuild.nvim/discussions/152

<div type='discussions-op-text'>

<sup>Originally posted by **Almaz5200** June 14, 2024</sup>
I have a modularized project that uses Tuist to generate projects. Because of that, each feature is a project in itself. However, when I add a new file it only shows me the targets from the root projects. I can still add them using `tuist generate`, but it would be a lot quicker if it could detect those targets as well. Is there any way to do that?</div>

**Comment by wojciech-kulik** [comment]
Partially resolved in #188

**Comment by Almaz5200** [comment]
@wojciech-kulik I wonder what's left to fully resolve this?

For me, the only thing left is editing Project.swift and some other Tuist helper files. For that purpose I think I can detect to which project the file belongs by just checking the filepaths. Those and only those files belong to Tuist that either are named Project.swift, or are in a "Tuist" folder in the current directory.

I guess that could be implemented by allowing to have multiple settings.json and buildServer.json files, as well as two LSP instances, but is running two LSP's looking at different buildServer.json's even possible?

**Comment by Almaz5200** [comment]
So, I've made something to fix my problem, let me know if you want some of it in the project. First, I have this function
``` lua
local Path = require('plenary.path')
local builtin = require('telescope.builtin')

function CopySelectedConfigFile()
  local config_dir = vim.fn.getcwd() .. "/.nvim/configurations"
  local dest_path = vim.fn.getcwd() .. "/.nvim/xcodebuild/settings.json"

  builtin.find_files({
    prompt_title = "Select a configuration",
    cwd = config_dir,
    attach_mappings = function(_, map)
      map('i', '<CR>', function(prompt_bufnr)
        local selected_file = require('telescope.actions.state').get_selected_entry().path
        require('telescope.actions').close(prompt_bufnr)

        -- Copy the selected file to the destination
        local source_path = Path:new(selected_file)
        local dest = Path:new(dest_path)
        source_path:copy({ destination = dest, recursive = false, overwrite = true })

        print("Copied " .. selected_file .. " to " .. dest_path)

        vim.cmd("XcodebuildReloadConfiguration")
      end)
      return true
    end,
  })
end
```
As you can see, it just takes one of the predetermined settings.json files, copies it to the correct place and runs XcodebuildReloadConfiguration, which I've wrote.
This command just reloads configuration from settings.json and re-runs xcode-build-server command.

I feel like project switching is too crude to incorporate like that, for one it doesn't account for changes in my two configurations, but XcodebuildReloadConfiguration maybe is good enough?

**Comment by wojciech-kulik** [comment]
@Almaz5200 I think we could close this issue. It was initially about being able to add a file to a specific `xcodeproj` which I believed is resolved by your PR: #188. 

Switching between projects is something different and could be tracked separately. What do you think?

**Comment by Almaz5200** [comment]
Yeah, it did actually grow out of it's initial purpose :D. So yeah, I think so too
